core-ui/src/main/java/com/maxshop/viewModel/MultiViewModuleFactory.kt

```diff
@@ -0,0 +1,18 @@
+package com.maxshop.viewModel
+
+import androidx.lifecycle.ViewModel
+import androidx.lifecycle.ViewModelProvider
+import javax.inject.Inject
+import javax.inject.Provider
+
+class MultiViewModuleFactory @Inject constructor(
+    private val viewModelFactory: Map<Class<out ViewModel>, @JvmSuppressWildcards Provider<ViewModel>>
+) : ViewModelProvider.Factory {
+
+    @Suppress
+    override fun <T : ViewModel> create(modelClass: Class<T>): T {
+        return viewModelFactory.getValue(modelClass as Class<ViewModel>).get() as T
+    }
+
+    val viewModelClasses get() = viewModelFactory.keys
```

Comment:
This variable is never used. For what purpose it exist? Should we remove it?

Reply:
fixed